pandas/_libs/lib.pyx

```diff
@@ -1324,12 +1330,12 @@ def infer_dtype(value: object, skipna: bool = True) -> str:
         # e.g. categoricals
         dtype = value.dtype
         if not isinstance(dtype, np.dtype):
-            value = _try_infer_map(value.dtype)
-            if value is not None:
-                return value
+            inferred = _try_infer_map(value.dtype)
+            if inferred is not None:
+                return inferred
 
             # its ndarray-like but we can't handle
-            raise ValueError(f"cannot infer type for {type(value)}")
+            values = np.asarray(value)
```

Comment:
if you are going to remove the exception then you can remove this L1337,1338 entirely (as np.asarray is called on L1341)

Reply:
> if you are going to remove the exception then you can remove this L1337,1338 entirely (as np.asarray is called on L1341)

I removed the duplicate `asarray`, but so we should still decide if we want to keep that exception or not